packages/unified-selection/src/unified-selection/Utils.ts

```diff
@@ -22,3 +23,37 @@ export function formIdBindings(property: string, ids: string[], bindings: ECSqlB
   ids.forEach((id) => bindings.push({ type: "id", value: id }));
   return `${property} IN (${ids.map(() => "?").join(",")})`;
 }
+
+/**
+ * Executes given ECSql query and extracts data from rows. Additionally handles main thread releasing.
+ * @internal
+ */
+export async function* executeQuery<T>(
+  queryExecutor: ECSqlQueryExecutor,
+  query: ECSqlQueryDef,
+  blockHandler: MainThreadBlockHandler,
```

Comment:
I don't think there's a need to pass the block handler from above - `executeQuery` can create one.